### ISSUE TITLE
Xcode 8 Support?

### ISSUE THREAD

**Issue by gbejarano01** [edwardaux/Ogra#28]
I was wondering if you're planning to upgrade the project so it's compatible with Xcode 8 and or Swift 2.3 / 3.0?

It would be great to be able to use this along with Argo.

Thanks!

**Comment by edwardaux** [comment]
yep, I just created a new `swift-2.3` branch and will merge #27 into there. 

I'll migrate `master` to be Swift3 to mirror the Argo projects plan. 

will leave this issue open until that's done.

**Comment by edwardaux** [comment]
Currently underway, but have hit a small wrinkle (https://github.com/thoughtbot/Argo/issues/412). Looking for workaround at the moment.

**Comment by jkennington** [comment]
While that fix is underway, if you need support for Swift 3, this can hold you over to just add to your project directly.

I haven't had time build this but there is a pull request on Carthage that may fix the issue. https://github.com/Carthage/Carthage/pull/1493

```
import Foundation
import Argo

public protocol Encodable {
    func encode() -> JSON
}

extension JSON: Encodable {
    public func encode() -> JSON {
        return self
    }
}

extension String: Encodable {
    public func encode() -> JSON {
        return .string(self)
    }
}

extension Bool: Encodable {
    public func encode() -> JSON {
        return .bool(self)
    }
}

extension Int: Encodable {
    public func encode() -> JSON {
        return .number(NSNumber(value: self))
    }
}

extension Double: Encodable {
    public func encode() -> JSON {
        return .number(NSNumber(value: self))
    }
}

extension Float: Encodable {
    public func encode() -> JSON {
        return .number(NSNumber(value: self))
    }
}

extension UInt: Encodable {
    public func encode() -> JSON {
        return .number(NSNumber(value: self))
    }
}

extension UInt64: Encodable {
    public func encode() -> JSON {
        return .number(NSNumber(value: self))
    }
}

extension Optional where Wrapped: Encodable {
    public func encode() -> JSON {
        switch self {
        case .none:        return .null
        case .some(let v): return v.encode()
        }
    }
}

extension Collection where Self: ExpressibleByDictionaryLiteral, Self.Key: ExpressibleByStringLiteral, Self.Value: Encodable, Iterator.Element == (Self.Key, Self.Value) {
    public func encode() -> JSON {
        var values = [String : JSON]()
        for (key, value) in self {
            values[String(describing: key)] = value.encode()
        }
        return .object(values)
    }
}

extension Optional where Wrapped: Collection & ExpressibleByDictionaryLiteral, Wrapped.Key: ExpressibleByStringLiteral, Wrapped.Value: Encodable, Wrapped.Iterator.Element == (Wrapped.Key, Wrapped.Value) {
    public func encode() -> JSON {
        return self.map { $0.encode() } ?? .null
    }
}

extension Collection where Iterator.Element: Encodable {
    public func encode() -> JSON {
        return JSON.array(self.map { $0.encode() })
    }
}

extension Optional where Wrapped: Collection, Wrapped.Iterator.Element: Encodable {
    public func encode() -> JSON {
        return self.map { $0.encode() } ?? .null
    }
}

extension Encodable where Self: RawRepresentable, Self.RawValue == String {
    public func encode() -> JSON {
        return .string(self.rawValue)
    }
}

extension Encodable where Self: RawRepresentable, Self.RawValue == Int {
    public func encode() -> JSON {
        return .number(NSNumber(value: self.rawValue))
    }
}

extension Encodable where Self: RawRepresentable, Self.RawValue == Double {
    public func encode() -> JSON {
        return .number(NSNumber(value: self.rawValue))
    }
}

extension Encodable where Self: RawRepresentable, Self.RawValue == Float {
    public func encode() -> JSON {
        return .number(NSNumber(value: self.rawValue))
    }
}

extension Encodable where Self: RawRepresentable, Self.RawValue == UInt {
    public func encode() -> JSON {
        return .number(NSNumber(value: self.rawValue))
    }
}

extension Encodable where Self: RawRepresentable, Self.RawValue == UInt64 {
    public func encode() -> JSON {
        return .number(NSNumber(value: self.rawValue))
    }
}

extension JSON {
    public func JSONObject() -> AnyObject {
        switch self {
        case .null:              return NSNull()
        case .string(let value): return value as AnyObject
        case .number(let value): return value
        case .array(let array):  return array.map { $0.JSONObject() } as AnyObject
        case .bool(let value):   return value as AnyObject
        case .object(let object):
            var dict: [Swift.String : AnyObject] = [:]
            for (key, value) in object {
                dict[key] = value.JSONObject()
            }
            return dict as AnyObject
        }
    }
}
```

**Comment by Eke** [comment]
Hello, any updates?

**Comment by edwardaux** [comment]
hi all... published 4.0.0 of Ogra tonight, which co-relates to Argo 4.0.0 including Swift 3.0 support. Enjoy... apologies for the delay.  Let me know if there are any problems.
